feat: set input.pull batch limit to 20000

stone-lyl opened the pull request:
- feat: set `input.pull` batch limit to 20000

- feat: replace eval with createJSTransformFunction/createJSTransformExpression to enhance performance for large datasets [link](https://jsbenchmark.com/#eyJjYXNlcyI6W3siaWQiOiJIRDVyWmZLVEdJeWpWNVotaGdDc2MiLCJjb2RlIjoiZXZhbCgnaXRlbSA9PiBPYmplY3QuYXNzaWduKHt9LCBpdGVtKScpKHtCOiAyLCBDOiAzLCBhOiAxfSk7IiwibmFtZSI6ImV2YWwiLCJkZXBlbmRlbmNpZXMiOltdfSx7ImlkIjoiSVFmN2JadlJIOFhLdG1RVEtaQmtEIiwiY29kZSI6IkRBVEEoJ2l0ZW0gPT4gT2JqZWN0LmFzc2lnbih7fSwgaXRlbSknKSh7QjogMiwgQzogMyxhOjF9KSIsIm5hbWUiOiJuZXcgRnVuY3Rpb24iLCJkZXBlbmRlbmNpZXMiOltdfV0sImNvbmZpZyI6eyJuYW1lIjoiQmFzaWMgZXhhbXBsZSIsInBhcmFsbGVsIjp0cnVlLCJnbG9iYWxUZXN0Q29uZmlnIjp7ImRlcGVuZGVuY2llcyI6W119LCJkYXRhQ29kZSI6ImNvbnN0IGZ1bmN0aW9uQ2FjaGUgPSBuZXcgTWFwPHN0cmluZywgV2Vha1JlZjxGdW5jdGlvbj4-KCk7XG5jb25zdCBjcmVhdGVKU1RyYW5zZm9ybUZ1bmN0aW9uID0gKGNvZGU6IHN0cmluZykgPT4ge1xuICBjb25zdCBjYWNoZWRGbiA9IGZ1bmN0aW9uQ2FjaGUuZ2V0KGNvZGUpPy5kZXJlZigpO1xuICBpZiAoY2FjaGVkRm4pIHJldHVybiBjYWNoZWRGbjtcblxuICBjb25zdCBmbiA9IG5ldyBGdW5jdGlvbignaXRlbScsIGByZXR1cm4gKCR7Y29kZX0pKGl0ZW0pYCk7XG4gIGZ1bmN0aW9uQ2FjaGUuc2V0KGNvZGUsIG5ldyBXZWFrUmVmKGZuKSk7XG4gIHJldHVybiBmbjtcbn07XG5yZXR1cm4gY3JlYXRlSlNUcmFuc2Zvcm1GdW5jdGlvbjsifX0)

- fix: resolve issue where the table Node fails to load data except on the first run

before：

https://github.com/user-attachments/assets/40053208-f35b-4047-aeac-212881e202a5

after:

https://github.com/user-attachments/assets/874709dd-c45d-4c46-8c8b-5f5cedaf19be

ajthinking commented on packages/core/src/computers/Map.ts:
```diff
@@ -18,9 +18,7 @@ export const Map: Computer = {
-        item => ({
-          ...item
-        })
+        item => Object.assign({}, item)
```
Though this is more performant, lets keep the example as the more straightforward spread syntax and let the user change that to Object.assign if needed

ajthinking commented on packages/ui/src/components/Node/table/UseObserverTable.tsx:
```diff
@@ -88,15 +85,14 @@ export function useObserverTable({ id, setIsDataFetched, setItems, items, parent
       onReceive: (linkIds) => {
         // If we have enough initial items, attempt unsubscribe
         if (itemsRef.current.length >= initialScreenCount) {
-          subscription?.unsubscribe();
```
Assuming this was due to "fix: resolve issue where the table Node fails to load data except on the first run" Howcome it did not work?

stone-lyl replied:
1. The first time we run `Run`, everything works fine.
2. However, when the condition itemsRef.current.length >= initialScreenCount is met, unsubscribe() is called.
3. This means the subscription is completely canceled.
4. As a result, even if we run `Run` again later, the onReceive callback will not be triggered since the previous subscription has been canceled.
5. This leads to `loadMore.current()` being unable to fetch new data subsequently. 
React expects components to re-render multiple times, so the correct approach is to let React's useEffect cleanup mechanism handle the subscription cleanup.

ajthinking replied:
Ok! Good call removing it. It was not a remarkable performance win anyway 👍

ajthinking reviewed (changes requested):
Looking more closely, I think we need to do some changes. This implementation breaks some nodes, for instance Merge.ts. That could be addressed, but im leaning towards we do it the other way around, that is - keep the precautions at the _Computer side_.

* Keep ability to do unbounded .pull() as well as pull(N)
* Instead move the "20000" number to relevant implementation sites, for instance in Map.ts etc.

stone-lyl commented:
> Looking more closely, I think we need to do some changes. This implementation breaks some nodes, for instance Merge.ts. That could be addressed, but im leaning towards we do it the other way around, that is - keep the precautions at the _Computer side_.
> 
> * Keep ability to do unbounded .pull() as well as pull(N)
> * Instead move the "20000" number to relevant implementation sites, for instance in Map.ts etc.


👍  first, I will address the issue in `Merge Node`. 
Then, I will add the batch limit on the computer side. 
I will find the batch limits for each node in `input.pull`.

stone-lyl commented:
1. I've implemented the `BatchLimit` on the computer side and added `BatchLimit = 20000` in nodes with `input.pull()`, excluding those with `canRun`.
2. I conducted sequential tests, and everything runs smoothly without causing any crashes in the VSCode extension.
3. After adding the `BatchLimit` to some nodes, like Map, I tested the Merge Node and found no further issues.

ajthinking reviewed (approved):
Great work 👍
